olp-cpp-sdk-core/include/olp/core/client/CancellationContext.h

```diff
@@ -50,6 +52,24 @@ class CORE_API CancellationContext {
 
   CancellationContext();
 
+    /**
+   * @brief Executes the given cancellable code block if the operation is not
+   * cancelled.
+   *
+   * @param execute_fn The function that should be executed if this operation is
+   * not cancelled. This function should return `CancellationToken` for which
+   * `CancellationContext` propagates a cancel request.
+   *
+   * Note: this method does not handle the case when context is alredy canceled
+   * and may produce unexpected results. Please prefer to use ExecuteOrCancelled
+   * with two arguments.
+   *
+   * @return True if the execution is successful; false if the context was
+   * cancelled.
+   */
+  OLP_SDK_DEPRECATED("Will be removed in 06.2021")
```

Comment:
I don't think this is necessary. We might need this one day or the user needs it and then we need to add it back. It is up to the user to use what is given in a senseable way.